TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -46,116 +49,78 @@ class BenchmarkParser extends Parser {
 
         while (curItr.done !== true) {
 
-            let curBenchmarkName = null;
-            let curBenchmarkVariant = null;
-            let curBenchmarkProduct = null;
-            let curBenchmarkNamedRouted = null;
+          
             let curMetric = null;
             let curSearchString = null;
             let curRegexResult = null;
             let curMetricValue = null;
-            let curJavaBuildDate = null;
-            let curJavaVersion = null;
             let isValid = true;
             let curTestData = {};
 
-            // Parse benchmark name
-            if ( ( curRegexResult = benchmarkNameRegex.exec( curItr.value ) ) !== null ) {
-                curBenchmarkName = curRegexResult[1];
-            } else {
-                isValid = false;
-            }
-
-            // Parse benchmark variant
-            if ( ( curRegexResult = benchmarkVariantRegex.exec( curItr.value ) ) !== null ) {
-                curBenchmarkVariant = curRegexResult[1];
-            } else {
-                isValid = false;
-            }
-
-            if (!BenchmarkMetricRouter[curBenchmarkName]) {
-                isValid = false;
-            } else if (!BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]) {
-                isValid = false;
-            // Benchmark should have at least one metric to parse with
-            } else if (!BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]
-            || !BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]
-            || !(Array.isArray(BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]))) {
-                isValid = false;
-            }
+            // parse product & benchmark
+            let curBenchmarkName = benchmarkNameRegex.exec( curItr.value )[1];
```

Comment:
I think we need the null check or else we'll crash later. Even though these snippets are automated in PerfNext, but we still need to be careful in case it is not elsewhere.